juspay/hyperswitch
# fix(connector): [Stripe] Deserialization Error while parsing Dispute Webhook Body

## Type of Change



## Description

`meta_data` struct is not present in case of dispute resulting in error during deserialization.

### Additional Changes





## Motivation and Context



## How did you test it?
Steps to test:

Through Postman
- Update merchant account with webhook url to recieve webhooks
- Update Payment connector(Stripe) with `merchant_secret`

In Stripe dashboard, update the webhook url to hyperswitch's webhook endpoint with formant
```
{{base_url}}/webhooks/{{merchant_id}}/stripe
```
<img width="1728" alt="Screenshot 2024-01-05 at 6 28 26 PM" src="https://github.com/juspay/hyperswitch/assets/73734619/58ec41e4-0b5d-4cd7-a5bc-586547b9ce30">

Different Events:
- Create a payment through Stripe:
`{
    "amount": 6540,
    "currency": "USD",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
    "setup_future_usage": "on_session",
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "4242424242424242",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "123"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    },
    "routing": {
        "type": "single",
        "data": "stripe"
    }
}`

Following are the webhook recieved through hyperswitch
<img width="1728" alt="Screenshot 2024-01-05 at 6 13 06 PM" src="https://github.com/juspay/hyperswitch/assets/73734619/9fbeff1e-1182-4387-955c-4e4e1521fce4">
<img width="1728" alt="Screenshot 2024-01-05 at 6 13 03 PM" src="https://github.com/juspay/hyperswitch/assets/73734619/cbe6acee-69e0-4c13-9de8-8962850d21af">

- Create a Refund Through Stripe:
`{
    "payment_id": "{{payment_id}}",
    "amount": 600,
    "reason": "Customer returned product",
    "refund_type": "instant",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}`

Following are the webhooks recieved through hyperswitch:
<img width="1728" alt="Screenshot 2024-01-05 at 6 15 36 PM" src="https://github.com/juspay/hyperswitch/assets/73734619/3f3e2122-288f-4110-87a5-50e08ae7cd93">


- Create a Dispute through Stripe:
`{
    "amount": 6540,
    "currency": "USD",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
    "setup_future_usage": "on_session",
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "4000000000000259",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "123"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    },
    "routing": {
        "type": "single",
        "data": "stripe"
    }
}`

Payment through this card will create a dispute option against that payment in Stripe Dashboard
Raise a counter or accept the dispute in dashboard to receive the webhooks.

<img width="1728" alt="Screenshot 2024-01-05 at 6 13 01 PM" src="https://github.com/juspay/hyperswitch/assets/73734619/7a6207fd-57c3-4303-909d-ad8afdf4e1cf">

<img width="1728" alt="Screenshot 2024-01-05 at 6 12 59 PM" src="https://github.com/juspay/hyperswitch/assets/73734619/c61ce9d2-9cd2-4bd3-b419-0996a5405733">

## Checklist
